libs/libc/stdlib/lib_cxa_atexit.c

```diff
@@ -0,0 +1,81 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_cxa_atexit.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/compiler.h>
+#include <nuttx/lib/lib.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include <sys/atexit.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif
+
+  FAR void *__dso_handle weak_data;
+
+#if defined(__cplusplus)
+}
+#endif
+
+FAR void *__dso_handle = &__dso_handle;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: __cxa_atexit
```

Comment:
need move __aeabi_atexit to libs/libc/machine/arm/?

Reply:
I don't know

Reply:
I saw you move libxx_cxa_atexit.cxx from libs/libxx/ to libs/libc/stdlib, then __aeabi_atexit should do the similar thing.

Reply:
Yes, cxa_atecit belongs to stdlib, but I don't know about aeabi_atexit as I don't know what it is. Something about the ARM ABI but I don't know why that is needed.

Reply:
Ok, I create a new patch for this: https://github.com/apache/incubator-nuttx/pull/6240. Please take a look.